src/backend/parser/parse_relation.c

```diff
@@ -3051,6 +3052,10 @@ expandNSItemAttrs(ParseState *pstate, ParseNamespaceItem *nsitem,
 							 (AttrNumber) pstate->p_next_resno++,
 							 label,
 							 false);
+
+		/* if transform * into columnlist with IMMV, remove IVM columns */
+		if (refStar && rte->relisivm && isIvmColumn(label) && !MatViewIncrementalMaintenanceIsEnabled())
+			continue;
```

Comment:
This refStar condition avoids to eliminate IVM columns in undesired situations, and this suppresses the following error which will occur if this condition doesn't exists.
```
postgres=# select __ivm_count__ from mv;
ERROR:  could not find attribute 2 in subquery targetlist
```

However, this hack would not be correct.  First, selectedCols are originally used for permission tests for columns, so I am not sure if this is usable in this manner. Second, this way causes the following wrong behaviour including Assert violations.

```sql
postgres=# select i,* from mv;
 i | i | __ivm_count__ 
---+---+---------------
(0 rows)

postgres=# select *,i from mv;
TRAP: FailedAssertion("dest_tle->resno == src_tle->resno", File: "tlist.c", Line: 351)

postgres=# select *,* from mv;
TRAP: FailedAssertion("dest_tle->resno == src_tle->resno", File: "tlist.c", Line: 351)
```

In my opinion, we firstly have to understand the root causal of "could not find attribute" error, then make a design for fixing this.